### PR TITLE
[VL] Replace std::cerr/cout with LOG and remove GLUTEN_PRINT_DEBUG macro

### PR DESCRIPTION
**Pull request by Yohahaha**
Replace all std::cerr/cout with LOG, replace DEBUG_OUT with DLOG, and remove GLUTEN_PRINT_DEBUG macro to avoid compilation issues caused by the lack of DEBUG testing.

### CODE REVIEW

**Comment by marin-ma** [comment]
Thanks. Would you rebase?

**Comment by marin-ma** [comment]
The find_package(glog REQUIRED) requires that the glog dependency is installed. Currently, glog is installed as part of building Velox dependencies. However, nowhere indicates that glog is a mandatory dependency for Gluten. 

As it's moved to top-level CMakeList.txt, I think it would be better if we can have something like
```
  find_package(glog ${GLUTEN_GLOG_MIN_VERSION} CONFIG)
  if(NOT glog_FOUND)
    include(BuildGlog)
  endif()
```

**Comment by Yohahaha** [comment]
yeah, I will do rebase later.

File: cpp/CMake/BuildGLog.cmake [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    "https://github.com/google/glog/archive/refs/tags/v${GLUTEN_GLOG_VERSION}.tar.gz"
+    )
+
+if(DEFINED ENV{GLUTEN_GLOG_SOURCE_URL})
+  set(GLUTEN_GLOG_SOURCE_URL "$ENV{GLUTEN_GLOG_SOURCE_URL}")
+endif()
+
+message(STATUS "Building glog from source")
+FetchContent_Declare(
+    gtest
```
Comment (marin-ma) [inline cpp/CMake/BuildGLog.cmake:17]: gtest -> glog

File: cpp/CMake/BuildGLog.cmake [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    URL ${GLUTEN_GLOG_SOURCE_URL}
+    URL_HASH "SHA256=${GLUTEN_GLOG_BUILD_SHA256_CHECKSUM}"
+)
+
+FetchContent_MakeAvailable(gtest)
```
Comment (marin-ma) [inline cpp/CMake/BuildGLog.cmake:22]: Same here.

**Comment by Yohahaha** [comment]
I dont know how to configure vcpkg correctly...

**Comment by marin-ma** [comment]
cc: @zhouyuan

**Comment by zhouyuan** [comment]
@Yohahaha @marin-ma  here's the place to add `glog ` dependency in vcpkg, If I understand this patch correctly
https://github.com/oap-project/gluten/blob/main/dev/vcpkg/vcpkg.json#L27-L42

thanks, -yuan

**Comment by marin-ma** [comment]
@Yohahaha I dig a bit deeper and found its probably caused by our Findglog.cmake module doesn't interface link to gflags.  https://github.com/oap-project/gluten/blob/ba045c7705d8fe6cd2a52d7b303a6a01fd54a8d8/cpp/CMake/Findglog.cmake#L33-L37

Let me open another PR to solve it first.

File: cpp/core/CMakeLists.txt [hunk 3]
```diff
@@ -325,7 +325,7 @@ else()
 endif()
 
 target_link_libraries(gluten
-    PUBLIC Arrow::arrow Arrow::parquet)
+    PUBLIC Arrow::arrow Arrow::parquet glog::glog)
```
Comment (marin-ma) [inline cpp/core/CMakeLists.txt:328]: The target found by our `Findglog` module is google::glog.

Comment (Yohahaha) [reply cpp/core/CMakeLists.txt]: thank you!

**Comment by Yohahaha** [comment]
```
OpenJDK 64-Bit Server VM warning: You have loaded library /tmp/gluten-be155c79-5c02-4f41-85ae-0a8c3c520afa/jni/0b25eb42-3ef8-4f32-a4b7-5dad99d25da5/gluten-2591252663942864927/libvelox.so which might have disabled stack guard. The VM will try to fix the stack guard now.
It's highly recommended that you fix the library with 'execstack -c <libfile>', or link it with '-z noexecstack'.
ERROR: something wrong with flag 'timestamp_in_logfile_name' in file '/opt/gluten/dev/vcpkg/.vcpkg/buildtrees/glog/src/v0.6.0-1e4356b0ac.clean/src/logging.cc'.  One possibility: file '/opt/gluten/dev/vcpkg/.vcpkg/buildtrees/glog/src/v0.6.0-1e4356b0ac.clean/src/logging.cc' is being linked both statically and dynamically into this executable.
```

I will pending this path since it is not important, and vcpkg env confused me a lot...

**Comment by marin-ma** [comment]
@Yohahaha This PR is very useful. Could you help to rebase?

File: cpp/core/CMakeLists.txt (hunk 3 above)
Comment (marin-ma) [reply cpp/core/CMakeLists.txt]: > The target found by our `Findglog` module is google::glog.

This change should be necessary.

**Comment by Yohahaha** [comment]
> @Yohahaha This PR is very useful. Could you help to rebase?

yeah, let me try again.

**Comment by Yohahaha** [comment]
@marin-ma could you help remove `stale` lable?

File: cpp/core/jni/JniCommon.h [hunk 4]
```diff
@@ -106,13 +105,11 @@ static inline jmethodID getStaticMethodIdOrError(JNIEnv* env, jclass thisClass,
 static inline void attachCurrentThreadAsDaemonOrThrow(JavaVM* vm, JNIEnv** out) {
   int getEnvStat = vm->GetEnv(reinterpret_cast<void**>(out), jniVersion);
   if (getEnvStat == JNI_EDETACHED) {
-    DEBUG_OUT << "JNIEnv was not attached to current thread." << std::endl;
```
Comment (marin-ma) [inline cpp/core/jni/JniCommon.h:109]: Please do not remove existing logs, unless for cleanup purposes. I would suggest replacing with `DLOG(INFO)`.

**Comment by marin-ma** [comment]
Would you please include the replacement policies in the PR description? e.g. Replacing `std::cerr` with `LOG(WARNING)`, replacing `GLUTEN_PRINT_DEBUG` with xxx, etc.

And please do not remove any existing debug logs. If you cleanup or removal is necessary, let's address that separately in another PR.

File: cpp/velox/jni/VeloxJniWrapper.cc [hunk 5]
```diff
@@ -49,8 +49,6 @@ jint JNI_OnLoad(JavaVM* vm, void*) {
   gluten::initVeloxJniFileSystem(env);
   gluten::initVeloxJniUDF(env);
 
-  DEBUG_OUT << "Loaded Velox backend." << std::endl;
```
Comment (marin-ma) [inline cpp/velox/jni/VeloxJniWrapper.cc:52]: Please do not remove it.

File: cpp/core/utils/qpl/qpl_job_pool.cc [hunk 6]
```diff
@@ -37,7 +37,7 @@ QplJobHWPool& QplJobHWPool::GetInstance() {
 QplJobHWPool::QplJobHWPool() : randomEngine(std::random_device()()), distribution(0, MAX_JOB_NUMBER - 1) {
   uint64_t initTime = 0;
   TIME_NANO(initTime, InitJobPool());
-  DEBUG_OUT << "Init job pool took " << 1.0 * initTime / 1e6 << "ms" << std::endl;
+  LOG(INFO) << "Init job pool took " << 1.0 * initTime / 1e6 << "ms";
```
Comment (marin-ma) [inline cpp/core/utils/qpl/qpl_job_pool.cc:40]: DLOG(INFO)

File: cpp/core/utils/qpl/qpl_job_pool.cc [hunk 7]
```diff
@@ -96,7 +96,7 @@ qpl_job* QplJobHWPool::AcquireJob(uint32_t& jobId) {
     }
   }
   jobId = MAX_JOB_NUMBER - index;
-  DEBUG_OUT << "Acquired job index " << index << " after " << retry << " retries." << std::endl;
+  LOG(INFO) << "Acquired job index " << index << " after " << retry << " retries.";
```
Comment (marin-ma) [inline cpp/core/utils/qpl/qpl_job_pool.cc:99]: DLOG(INFO)

**Review by marin-ma — COMMENTED** [review]
LGTM. Thank you for tracking this PR! Just a few minor comments.

File: cpp/velox/jni/VeloxJniWrapper.cc (hunk 5 above)
Comment (Yohahaha) [reply cpp/velox/jni/VeloxJniWrapper.cc:52]: ok.

**Comment by Yohahaha** [comment]
@marin-ma @rui-mo please help review again, thanks!

**Comment by marin-ma** [comment]
LGTM. Thanks for iterating!

**Review by marin-ma — APPROVED** [review]

File: cpp/velox/compute/WholeStageResultIterator.cc [hunk 8]
```diff
@@ -237,8 +237,8 @@ void WholeStageResultIterator::collectMetrics() {
     const auto& nodeId = orderedNodeIds_[idx];
     if (planStats.find(nodeId) == planStats.end()) {
       if (omittedNodeIds_.find(nodeId) == omittedNodeIds_.end()) {
-        DEBUG_OUT << "Not found node id: " << nodeId << std::endl;
-        DEBUG_OUT << "Plan Node: " << std::endl << veloxPlan_->toString(true, true) << std::endl;
+        LOG(WARNING) << "Not found node id: " << nodeId;
+        LOG(WARNING) << "Plan Node: " << std::endl << veloxPlan_->toString(true, true);
```
Comment (rui-mo) [inline cpp/velox/compute/WholeStageResultIterator.cc:241]: > replace DEBUG_OUT with DLOG

Seems not matched with this description. Any reason here?

File: cpp/core/jni/JniWrapper.cc [hunk 9]
```diff
@@ -92,11 +91,10 @@ class JavaInputStreamAdaptor final : public arrow::io::InputStream {
     try {
       auto status = JavaInputStreamAdaptor::Close();
       if (!status.ok()) {
-        DEBUG_OUT << __func__ << " call JavaInputStreamAdaptor::Close() failed, status:" << status.ToString()
-                  << std::endl;
+        LOG(WARNING) << __func__ << " call JavaInputStreamAdaptor::Close() failed, status:" << status.ToString();
       }
     } catch (std::exception& e) {
-      DEBUG_OUT << __func__ << " call JavaInputStreamAdaptor::Close() got exception:" << e.what() << std::endl;
+      LOG(WARNING) << __func__ << " call JavaInputStreamAdaptor::Close() got exception:" << e.what();
```
Comment (rui-mo) [inline cpp/core/jni/JniWrapper.cc:97]: Why not `DLOG` here?

File: cpp/velox/compute/WholeStageResultIterator.cc (hunk 8 above)
Comment (marin-ma) [reply cpp/velox/compute/WholeStageResultIterator.cc:241]: Changing the log type to warning in the exception block does make sense to me.

File: cpp/core/jni/JniWrapper.cc (hunk 9 above)
Comment (Yohahaha) [reply cpp/core/jni/JniWrapper.cc:97]: exception msg should be warning.

**Review by rui-mo — APPROVED** [review]
